feat(view): add Network Syncs tab

lonelybud opened the pull request:
refractor code
- changes in VIEW_PLAYER_INFO_EXTRA_INFO
  - add block_ptfx option
  - display full vehicle name
- changes in gta_data_service
  - change key type in vehicle_map & ped_map
  - add get_vehicle_full_name
 
added GUI_NETWORK_SYNCS tab to track ped, vehicle, pickups, objects, explosion, ptfx events


closes #3369, #1373, #3273

lonelybud committed:
feat(debug): log explosion event

Yimura commented on src/hooks/protections/received_event.cpp:
```diff
@@ -867,7 +881,15 @@ namespace big
-			scan_explosion_event(source_player, buffer);
+			scan_explosion_event(plyr, source_player, buffer);
+
+			// dont accept event in died state
```
Don't let the game handle the event if we're dead you mean? We're still accepting the event (from the attackers POV).

lonelybud replied:
if we are dead the explosion event is useless to us, it will only create explosions around the dead camera scene if intended for us. Also i have added the check later so that the logging can still happen. I am open to your ideas.

lonelybud replied:
hmn i think it will create problem since it is not specific. Some explosion like car being exploded etc which are legit will also get rejected. This is concerning you are right

Yimura replied:
I was more targeting the wording used in the comment, we're really still accepting the event just not letting the game process it and create the explosion on our end.

lonelybud replied:
right we are just sending the acknowledgement but in game we are not processing the explosion. I have not tested this with scenarios like when you die and someone else die in meantime bcz of explosion or someone's car get exploded. Therefore i removed it.

Yimura commented on src/hooks/protections/received_event.cpp:
```diff
@@ -386,6 +387,19 @@ namespace big
 
+		// logs all type of explosion including owned one, fire, water hydrant etc, one without damage but camerashake, npcs shooting explosive ammo from planes from source client etc
+		if (g.debug.logs.explosion_event && plyr)
+			{
```
Mixed indents

lonelybud committed:
fix indentation

lonelybud committed:
add proper descriptions

lonelybud committed:
remove condition where we dont accept event when died

lonelybud committed:
prevent log spam

lonelybud committed:
Merge branch 'master' into log_explosion_event

lonelybud committed:
refractor gta_data_service & VIEW_PLAYER_INFO_EXTRA_INFO

- changes in VIEW_PLAYER_INFO_EXTRA_INFO
  - add block_ptfx option in
  - display full vehicle name
- changes in gta_data_service
  - change key type in vehicle_map & ped_map
  - add get_vehicle_full_name

lonelybud committed:
add VIEW_PLAYER_INFO_BLOCK_PTFX_EVENTS

lonelybud committed:
remove previous code & add GUI_NETWORK_SYNCS tab

- remove code related to logging explosion event in console
- added GUI_NETWORK_SYNCS tab to track ped, vehicle, pickups, objects, explosion, ptfx events

lonelybud committed:
Merge branch 'master' into log_explosion_event

lonelybud committed:
format files and do renames
